Formula/t/teleport.rb

```diff
@@ -43,6 +46,10 @@ class Teleport < Formula
   conflicts_with "tctl", because: "both install `tctl` binaries"
 
   def install
+    ENV.prepend_path "PATH", Formula["rustup"].bin
+    system "rustup", "default", "stable"
+    system "rustup", "set", "profile", "minimal"
```

Comment:
Do we really need `rustup`? If we're already using it, why not use `rust`?

Reply:
Probably since we don't build `wasm32-unknown-unknown` (https://rustwasm.github.io/wasm-pack/book/prerequisites/non-rustup-setups.html). Not sure if rustc can find it outside its own sysroot if we want to follow docs.

May be worth checking how difficult it is to build with `rust`, e.g. https://archlinux.org/packages/extra/x86_64/rust-wasm/

---

Also I think `corepack` downloads its own copy of `pnpm`. Could probably use our own rather than depending on `corepack`.

Reply:
Yea, I've been wondering about adding wasm support to our own `rust`, since that seems to be what's behind the remaining build/run-time usage for `rustup`.

Reply:
Yes, without rustup it was complaining about wasm32-unknown-unknown, so i added it.
There are probably other ways to address that, but I'd consider that as an optimisation after we have it working and updatable.

Reply:
Some optimisations can be tried in 
* https://github.com/Homebrew/homebrew-core/pull/191838